From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 9 Sep 2026 13:13:58 +0530
Subject: [PATCH 024/109] Cygwin: add dummy implementation of
 _fe_nomask_env

(cherry picked from commit 623eef974404d571a4cf7815d6363fe3341b6b7a)
---
 winsup/cygwin/fenv.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/winsup/cygwin/fenv.c b/winsup/cygwin/fenv.c
index 80f7cc52c..1558f76c2 100644
--- a/winsup/cygwin/fenv.c
+++ b/winsup/cygwin/fenv.c
@@ -3,3 +3,13 @@
    being called from mainCRTStartup in crt0.o. */
 void _feinitialise (void)
 {}
+
+#if defined(__aarch64__)
+
+#include <fenv.h>
+#include <stddef.h>
+
+/* _fe_nomask_env is exported by cygwin.din but not used at all for AArch64. */
+const fenv_t *_fe_nomask_env = NULL;
+
+#endif /* __aarch64__ */
